### PR TITLE
Feature/cors middleware v2

### PR DESCRIPTION
**Pull request by j0s**

### CODE REVIEW

**Comment by j0s** [comment]
Since this isn't technically a middleware, does it belong in some other repo?

File: cors/handler.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
+package cors
```
Comment (j0s) [inline cors/handler.go:1]: Maybe cors/preflight would be better? And then we could add an actual middleware that adds `Access-Control-Allow-Origin` in package cors

File: cors/preflight/handler.go [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+			w.Header().Set("Access-Control-Allow-Methods", allowedMethods)
+		}
+
+		if allowedHeaders := strings.Join(allowedHeaders, ", "); allowedHeaders != "" {
+			w.Header().Set("Access-Control-Allow-Headers", allowedHeaders)
+		}
+	}
+}
+
+func AddHandler(router *mux.Router, allowedHeaders ...string) {
```
Comment (Ekberg) [inline cors/preflight/handler.go:24]: Maybe we can leave this to the consuming API?

Because you should be able to differ the headers list per endpoint?

Comment (j0s) [reply cors/preflight/handler.go:24]: I just thought would be simplest for the API consumers, and I don't really see the problem with allowing some headers on endpoints that don't use them.

If we are going to allow different headers on each endpoint I guess we would need to make a `Preflight` struct, with methods like `AddAllowedHeaders(path string, headers []string)` or something similar? Or did you have another solution in mind?

**Comment by Ekberg** [comment]
Closing since we merged another alternative implementation

**Pull request closed without merging**